File: 1248. Count Number of Nice Subarrays.java
class Solution {
    public int numberOfSubarrays(int[] nums, int k) {
        int answer = 0, count = 0;
        Map<Integer, Integer> freq = new HashMap<>();
        freq.put(0, 1);
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] % 2 != 0) count++;
            answer += freq.getOrDefault(count-k, 0);
            freq.put(count, freq.getOrDefault(count, 0) + 1);
        }
        return answer;
    }
}

File: go/1248. Count Number of Nice Subarrays.go
func numberOfSubarrays(nums []int, k int) int {
    n := len(nums)
    freq := make(map[int]int)
    freq[0] = 1
    count := 0
    answer := 0
    for i := 0; i < n; i++ {
        if nums[i] % 2 != 0 {
            count += 1
        }
        answer += freq[count-k]
        freq[count] += 1
    }
    return answer
}
